### ISSUE TITLE
[508-defect-2] Screenreader experience a11y audit, Financial Status Report

### ISSUE THREAD

**Issue by AngelaFowler82** [department-of-veterans-affairs/va.gov-team#36116]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

Issue Title: 508 defect 2, Screen Readers, Financial Status Report, results of A11Y review
Point of Contact: Angela Fowler
**VFS Point of Contact:** _First name only: Angela

## Details
Below is a series of A11Y issues with the Financial Status Report form in the order in which they appear split between form system issues and FSR issues:

### Form system 
Highlighting these for documentation purposes. These are out of scope for FSR.
- Page titles should more accurately state the pages purpose. Titles of each page read "Financial Status Report," which does not tell the user where they are in the process. [Note from Josh: there are upstream problems related to the design of the form system where the `h1` repeats the name of the form as opposed to providing the _subject_ of the questions being posed on that specific page. I'll reach out to platform spike team about this!]
- Submission page: the name field gives me an error before I even start entering the name.

### FSR
These issues should be addressable by the team
- On each page, below the h2 which tells the user what step they are on, there is a legend indicating the subsection. The text of that legend should be an h3. Place the h3 inside the legend code.
- The heading in the “Are you sure you want to start over” modal should be an h1. On the VA design system, [the storybooks modal uses a `h1`](https://design.va.gov/components/modal).
- On the page "What debt do you need help with," the “What if my debt isn’t listed” should be an h2. Also, Checkboxes should state purpose and target, I.E. “Request help with post 9/11 GI bill debt.”
- Disability Compensation and Pension Benefits should be an h3.
- When asked to enter the amount of money received from Social Security, hint text should be provided to the effect of "do not change this if you do not receive social security. 
- Similarly, when asked to enter assets such as stocks, it is unclear that the user has to enter "0" if they do not own the assets. While there is text above the form stating as much, it can easily be missed by screen reader users or anyone in a hurry. If the user gets it wrong, they are presented with an error message which states "enter your financial information. The error can be restated to instruct the user to please enter a number and to enter 0 if they do not own the asset.
- Repayment or Relief options," "Your financial overview" should be an h3.
- Where you choose an option fo reach debt, the legends should state which debt they refer too. As it is, a screen reader user tabbing through the page will repeatedly here "Which optionwould you like for this debt."
- Personal statement page implies that it refers to only one debt, though I believe it refers to all debts included in the application. Wording should be changed to reflect this.

### Other
- On the “This application is already in progress” page, the h3 informing me that my application for financial hardship is in progress should be an h2. I also question the lengthy heading text. [Note from Josh: we updated this from a previous version that repeated information in the heading and text following it several times. I can create a ticket/look into kicking up another iteration to reduce the heading length.]

```[tasklist]
### Tasks
- [x] On each page, below the h2 which tells the user what step they are on, there is a legend indicating the subsection. The text of that legend should be an h3. Place the h3 inside the legend code.
- [ ] The heading in the “Are you sure you want to start over” modal should be an h1. On the VA design system, the storybooks modal uses a h1.
- [x] On the page "What debt do you need help with," the “What if my debt isn’t listed” should be an h2. Also, Checkboxes should state purpose and target, I.E. “Request help with post 9/11 GI bill debt.”
- [x] Disability Compensation and Pension Benefits should be an h3.
- [x] When asked to enter the amount of money received from Social Security, hint text should be provided to the effect of "do not change this if you do not receive social security.
- [x] Similarly, when asked to enter assets such as stocks, it is unclear that the user has to enter "0" if they do not own the assets. While there is text above the form stating as much, it can easily be missed by screen reader users or anyone in a hurry. If the user gets it wrong, they are presented with an error message which states "enter your financial information. The error can be restated to instruct the user to please enter a number and to enter 0 if they do not own the asset.
- [x] Repayment or Relief options," "Your financial overview" should be an h3.
- [x] Where you choose an option fo reach debt, the legends should state which debt they refer too. As it is, a screen reader user tabbing through the page will repeatedly here "Which option would you like for this debt."
- [x] Personal statement page implies that it refers to only one debt, though I believe it refers to all debts included in the application. Wording should be changed to reflect this.
- [x] On the “This application is already in progress” page, the h3 informing me that my application for financial hardship is in progress should be an h2. I also question the lengthy heading text. [Note from Josh: we updated this from a previous version that repeated information in the heading and text following it several times. I can create a ticket/look into kicking up another iteration to reduce the heading length.]
```

**Comment by joshkimux** [comment]
@Dr-Pongo @OptionSelect hi y'all 🌤️ , checking in on this ticket to see if these issues have been resolved or are in any upcoming sprint planning.

Context: we're putting together an [a11y dashboard to keep track of each product's a11y health](https://docs.google.com/spreadsheets/d/1lgzMiZZ-0rBPDuB2gRupBAjsyF0WTe005VvEHYrMxww/edit?usp=sharing). Platform recommends fixing `defect-2` (major accessibility) issues within 1-2 sprints upon launch (or in this case, discovery). 

Some paths forward:
- if y'all are no longer responsible for this ticket, would y'all be able to point us in the right direction to who might own this now?
- if y'all are responsible for this ticket,
  - aim to resolve this within the next 1-2 sprints or as soon as possible
  - get support from or ask any questions to the new CAIA (content, accessibility, IA) team at #accessibility-help on slack ❤️

**Comment by HeatherRienks** [comment]
Thanks @hdjustice!

**Comment by denisecoveyduc** [comment]
Thanks @hdjustice moving to fast it seems ;P

**Comment by HeatherRienks** [comment]
@joshkimux - as @denisecoveyduc noted above, our FE team is very focused on completing all of our FSR enhancements over the next few sprints, so we will need to address this as soon as that development is completed. We are aiming for mid-May, but we have prioritized this work in our "Ready" column, so we don't lose sight of it. Feel free to check in anytime with questions!

**Comment by HeatherRienks** [comment]
@josephrlee has completed several of these tasks and will comment on what has been done.

**Comment by josephrlee** [comment]
Just took a look at it! I think the majority of the issues here have already been addressed! It also looks like some of these issues are in reference to the old experience of the form and no longer apply! 

The only thing I need clarification on is the Modal issue, @Dr-Pongo, can you help me with that one?

All the headings are now correct, and all h3s are inside of legends now. I added a debt(s) to the Resolution Comments page and will submit a tiny PR for the fix.
